whois-nrtm4/src/main/java/net/ripe/db/nrtm4/SnapshotInitializer.java

```diff
@@ -0,0 +1,57 @@
+package net.ripe.db.nrtm4;
+
+import net.ripe.db.nrtm4.persist.NrtmSource;
+import net.ripe.db.nrtm4.persist.NrtmVersionInfo;
+import net.ripe.db.nrtm4.persist.NrtmVersionInfoRepository;
+import net.ripe.db.nrtm4.persist.SnapshotObjectRepository;
+import net.ripe.db.whois.common.dao.SerialDao;
+import net.ripe.db.whois.common.domain.serials.SerialEntry;
+import net.ripe.db.whois.common.rpsl.Dummifier;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+
+@Service
+public class SnapshotInitializer {
+
+    private final NrtmVersionInfoRepository nrtmVersionInfoRepository;
+    private final SerialDao serialDao;
+    private final SnapshotObjectRepository snapshotObjectRepository;
+    private final Dummifier dummifierNrtm;
+
+    SnapshotInitializer(
+        final NrtmVersionInfoRepository nrtmVersionInfoRepository,
+        final SerialDao serialDao,
+        final SnapshotObjectRepository snapshotObjectRepository,
+        final Dummifier dummifierNrtm
+    ) {
+        this.nrtmVersionInfoRepository = nrtmVersionInfoRepository;
+        this.serialDao = serialDao;
+        this.snapshotObjectRepository = snapshotObjectRepository;
+        this.dummifierNrtm = dummifierNrtm;
+    }
+
+    NrtmVersionInfo init(final NrtmSource source) {
+        final List<SerialEntry> objects = serialDao.getSerialEntriesFromLast().stream()
+            .filter((obj) -> dummifierNrtm.isAllowed(NrtmConstants.NRTM_VERSION, obj.getRpslObject()))
```

Comment:
If it's a initialisation of snapshot there will be thousands of objects, do you need a streamer or may be partition?

Reply:
yes, i'll try a stream.